File: flight_service/src/common/logger.ts
import { settings } from "../config/settings.js";
import { Colors } from "./constants.js";

export class Logger {
    name: string;
    debugMode: boolean;

    constructor(name: string) {
        this.name = name;
        this.debugMode = settings.DEBUG_MODE;
    }

    info = (message: any) => {
        console.log(`${this.getPrefix('INFO', Colors.GREEN)} ${this.formatMessage(message)}`);
    }

    error = (message: any) => {
        console.log(`${this.getPrefix('ERROR', Colors.RED)} ${this.formatMessage(message)}`);
    }

    warn = (message: any) => {
        console.log(`${this.getPrefix('WARN', Colors.YELLOW)} ${this.formatMessage(message)}`);
    }

    debug = (message: any) => {
        if (this.debugMode) {
            console.log(`${this.getPrefix('DEBUG', Colors.CYAN)} ${this.formatMessage(message)}`);
        }
    }

    private getPrefix(level: string, color: string): string {
        const timestamp = new Date().toISOString().split('T')[1]?.split('Z')[0]; // HH:mm:ss.ms
        const fixedName = this.name.padEnd(20, ' '); 
        const fixedLevel = level.padEnd(7, ' ');    
        
        return `${Colors.DIM}${timestamp}${Colors.RESET} ${color}${fixedLevel}${Colors.RESET} ${Colors.MAGENTA}[${fixedName}]${Colors.RESET}`;
    }

    private formatMessage(message: any): string {
        return typeof message === 'object' 
            ? `\n${JSON.stringify(message, null, 2)}` 
            : message;
    }

}

File: airport_service/src/core/logging/logger.py
from datetime import datetime
from typing   import Any


class Logger:
    COLORS = {
        "DEBUG": "\033[36m",     # Cyan
        "INFO": "\033[32m",      # Green
        "WARNING": "\033[33m",   # Yellow
        "ERROR": "\033[31m",     # Red
        "CRITICAL": "\033[41m",  # Red background
    }

    RESET = "\033[0m"

    def __init__(self, name: str, level: str = "INFO"):
        self.name = name
        self.level = level

    def _log(self, level: str, message: str, **kwargs: Any) -> None:
        timestamp = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
        color = self.COLORS.get(level, "")

        extra = ""
        if kwargs:
            extra = " | " + " ".join(f"{k}={v}" for k, v in kwargs.items())

        print(
            f"{color}{timestamp} | {level:<8} | {self.name} | {message}{extra}{self.RESET}"
        )

    def debug(self, message: str, **kwargs: Any) -> None:
        self._log("DEBUG", message, **kwargs)

    def info(self, message: str, **kwargs: Any) -> None:
        self._log("INFO", message, **kwargs)

    def warning(self, message: str, **kwargs: Any) -> None:
        self._log("WARNING", message, **kwargs)

    def error(self, message: str, **kwargs: Any) -> None:
        self._log("ERROR", message, **kwargs)

    def critical(self, message: str, **kwargs: Any) -> None:
        self._log("CRITICAL", message, **kwargs)
